tests/integration/test_proxy.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for Flask charm proxy setting."""
+
+import requests
+from juju.model import Model
+from pytest import Config
+
+
+async def test_proxy(build_charm: str, model: Model, pytestconfig: Config, get_unit_ips):
+    """Build and deploy the flask charm."""
+    app_name = "flask-k8s"
+    await model.set_config(
+        {
+            "juju-http-proxy": "http://proxy.test",
+            "juju-https-proxy": "http://proxy.test",
+            "juju-no-proxy": "127.0.0.1,10.0.0.1",
+        }
+    )
+    resources = {
+        "flask-app-image": pytestconfig.getoption("--test-flask-image"),
```

Comment:
Should this come from a fixture and include an error message if the dev didn't supply it?

Reply:
Added, thanks!